File: Support Material/Demofiles/M5/CS/CustomerManagementEntities/Customer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CustomerManagementEntities
{
    public class Customer
    {
        private Guid? customerID;
        private string customerName = null;
        private string customerAddress = null;
        private string customerPhone = null;
        private string customerZipCode = null;
        private string customerCity = null;
        private string customerState = null;
        private Guid? customerCountryID = null;
        private string customerEmailAddress = null;
        private int customerCreditLimit = 0;
        private DateTime? customerCreatedDate = null;
        private string customerCreatedBy = null;
        private DateTime? customerModifiedDate = null;
        private string customerModifiedBy = null;

        /// <summary>
        /// The unique customer ID
        /// </summary>
        public Guid? ID
        {
            get
            {
                return this.customerID;
            }
            private set
            {
                this.customerID = value;
            }
        }

        /// <summary>
        /// The full customer name
        /// </summary>
        public string Name
        {
            get
            {
                return this.customerName;
            }
            set
            {
                // Null value?
                if (value == null)
                    this.customerName = "";
                else
                    // Only get the first 50 characters
                    if (value.Length > 50)
                        this.customerName = value.Substring(0, 50);
                    else
                        this.customerName = value;
            }
        }

        /// <summary>
        /// The customer address, including street name, house number and floor
        /// </summary>
        public string Address
        {
            get
            {
                return this.customerAddress;
            }
            set
            {
                // Null value?
                if (value == null)
                    this.customerAddress = "";
                else
                    // Only get the first 50 characters
                    if (value.Length > 50)
                        this.customerAddress = value.Substring(0, 50);
                    else
                        this.customerAddress = value;
            }
        }

        /// <summary>
        /// The customer phone number
        /// </summary>
        public string Phone
        {
            get
            {
                return this.customerPhone;
            }
            set
            {
                // Null value?
                if (value == null)
                    this.customerPhone = "";
                else
                    // Only get the first 30 characters
                    if (value.Length > 30)
                        this.customerPhone = value.Substring(0, 30);
                    else
                        this.customerPhone = value;
            }
        }

        /// <summary>
        /// The customer zip code or postal code
        /// </summary>
        public string ZipCode
        {
            get
            {
                return this.customerZipCode;
            }
            set
            {
                // Null value?
                if (value == null)
                    this.customerZipCode = "";
                else
                    // Only get the first 10 characters
                    if (value.Length > 10)
                        this.customerZipCode = value.Substring(0, 10);
                    else
                        this.customerZipCode = value;
            }
        }

        /// <summary>
        /// The name of the city in which the customer lives
        /// </summary>
        public string City
        {
            get
            {
                return this.customerCity;
            }
            set
            {
                // Null value?
                if (value == null)
                    this.customerCity = "";
                else
                    // Only get the first 30 characters
                    if (value.Length > 30)
                        this.customerCity = value.Substring(0, 30);
                    else
                        this.customerCity = value;
            }
        }

        /// <summary>
        /// The name of the state or region in which the customer lives
        /// </summary>
        public string State
        {
            get
            {
                return this.customerState;
            }
            set
            {
                // Null value?
                if (value == null)
                    this.customerState = "";
                else
                    // Only get the first 30 characters
                    if (value.Length > 30)
                        this.customerState = value.Substring(0, 30);
                    else
                        this.customerState = value;
            }
        }

        /// <summary>
        /// The ID of the country in which the customer lives
        /// </summary>
        public Guid? CountryID
        {
            get
            {
                return this.customerCountryID;
            }
            set
            {
                this.customerCountryID = value;
            }
        }

        /// <summary>
        /// The customer e-mail address
        /// </summary>
        public string EmailAddress
        {
            get
            {
                return this.customerEmailAddress;
            }
            set
            {
                // Null value?
                if (value == null)
                    this.customerEmailAddress = "";
                else
                    // Only get the first 40 characters
                    if (value.Length > 40)
                        this.customerEmailAddress = value.Substring(0, 40);
                    else
                        this.customerEmailAddress = value;
            }
        }

        /// <summary>
        /// The current credit limit of the customer
        /// </summary>
        public int CreditLimit
        {
            get
            {
                return this.customerCreditLimit;
            }
            set
            {
                // Negative value?
                if (value < 0)
                    this.customerCreditLimit = 0;
                else
                    this.customerCreditLimit = value;
            }
        }

        /// <summary>
        /// The date the customer was created in the system
        /// </summary>
        public DateTime? CreatedDate
        {
            get
            {
                return this.customerCreatedDate;
            }
            private set
            {
                // Date in the past?
                if (value < DateTime.Now)
                    this.customerCreatedDate = DateTime.Now;
                else
                    this.customerCreatedDate = value;
            }
        }

        /// <summary>
        /// The name of the user creating the customer
        /// </summary>
        public string CreatedBy
        {
            get
            {
                return this.customerCreatedBy;
            }
            set
            {
                // Null value?
                if (value == null)
                    this.customerCreatedBy = "";
                else
                    // Only get the first 15 characters
                    if (value.Length > 15)
                        this.customerCreatedBy = value.Substring(0, 15);
                    else
                        this.customerCreatedBy = value;
            }
        }

        /// <summary>
        /// The date the customer was last modified in the system
        /// </summary>
        public DateTime? ModifiedDate
        {
            get
            {
                return this.customerModifiedDate;
            }
            set
            {
                // Date in the past?
                if (value < DateTime.Now)
                    this.customerModifiedDate = DateTime.Now;
                else
                    this.customerModifiedDate = value;
            }
        }

        /// <summary>
        /// The name of the user who last modified the customer
        /// </summary>
        public string ModifiedBy
        {
            get
            {
                return this.customerModifiedBy;
            }
            set
            {
                // Null value?
                if (value == null)
                    this.customerModifiedBy = "";
                else
                    // Only get the first 15 characters
                    if (value.Length > 15)
                        this.customerModifiedBy = value.Substring(0, 15);
                    else
                        this.customerModifiedBy = value;
            }
        }

        /// <summary>
        /// Default parameterless constructor
        /// </summary>
        public Customer()
        {
            // Initialize member backing fields with default values
            this.ID = Guid.NewGuid();
            this.CreatedDate = DateTime.Now;
        }

        public Customer(Guid? id)
        {
            // Initialize member backing fields with 
            // passed and default values
            this.ID = id;
            this.CreatedDate = DateTime.Now;
        }

        /// <summary>
        /// Initializes with a value for all backing fields 
        /// </summary>
        /// <param name="id"></param>
        /// <param name="name"></param>
        /// <param name="address"></param>
        /// <param name="phone"></param>
        /// <param name="zipCode"></param>
        /// <param name="city"></param>
        /// <param name="state"></param>
        /// <param name="countryID"></param>
        /// <param name="emailAddress"></param>
        /// <param name="creditLimit"></param>
        /// <param name="createdDate"></param>
        /// <param name="createdBy"></param>
        /// <param name="modifiedDate"></param>
        /// <param name="modifiedBy"></param>
        public Customer(Guid? id, string name, string address, string phone,
            string zipCode, string city, string state, Guid? countryID,
            string emailAddress, int creditLimit, DateTime? createdDate,
            string createdBy, DateTime? modifiedDate, string modifiedBy)
        {
            // Initialize member backing fields with passed values
            this.ID = id;
            this.CreatedDate = DateTime.Now;
            this.Name = name;
            this.Address = address;
            this.Phone = phone;
            this.ZipCode = zipCode;
            this.City = city;
            this.State = state;
            this.CountryID = countryID;
            this.EmailAddress = emailAddress;
            this.CreditLimit = creditLimit;
            this.CreatedDate = createdDate;
            this.CreatedBy = createdBy;
            this.ModifiedDate = modifiedDate;
            this.ModifiedBy = modifiedBy;
        }
    }
}

File: Support Material/Demofiles/M5/VB/CustomerManagementEntities/Customer.vb
Public Class Customer
    Private customerID As Guid?
    Private customerName As String = Nothing
    Private customerAddress As String = Nothing
    Private customerPhone As String = Nothing
    Private customerZipCode As String = Nothing
    Private customerCity As String = Nothing
    Private customerState As String = Nothing
    Private customerCountryID? As Guid = Nothing
    Private customerEmailAddress As String = Nothing
    Private customerCreditLimit As Integer = 0
    Private customerCreatedDate? As DateTime = Nothing
    Private customerCreatedBy As String = Nothing
    Private customerModifiedDate? As DateTime = Nothing
    Private customerModifiedBy As String = Nothing

    ''' <summary>
    ''' The unique customer ID
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property ID() As Guid?
        Get
            Return Me.customerID
        End Get
        Private Set(ByVal value As Guid?)
            Me.customerID = value
        End Set
    End Property

    ''' <summary>
    ''' The full customer name
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property Name() As String
        Get
            Return Me.customerName
        End Get
        Set(ByVal value As String)
            ' Null value?
            If value Is Nothing Then
                Me.customerName = ""
            Else
                ' Only get the first 50 characters
                If (value.Length > 50) Then
                    Me.customerName = value.Substring(0, 50)
                Else
                    Me.customerName = value
                End If
            End If
        End Set
    End Property

    ''' <summary>
    ''' The customer address, including street name, house number and floor
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property Address() As String
        Get
            Return Me.customerAddress
        End Get
        Set(ByVal value As String)
            ' Null value?
            If value Is Nothing Then
                Me.customerAddress = ""
            Else
                ' Only get the first 50 characters
                If (value.Length > 50) Then
                    Me.customerAddress = value.Substring(0, 50)
                Else
                    Me.customerAddress = value
                End If
            End If
        End Set
    End Property

    ''' <summary>
    ''' The customer phone number
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property Phone() As String
        Get
            Return Me.customerPhone
        End Get
        Set(ByVal value As String)
            ' Null value?
            If value Is Nothing Then
                Me.customerPhone = ""
            Else
                ' Only get the first 30 characters
                If (value.Length > 30) Then
                    Me.customerPhone = value.Substring(0, 30)
                Else
                    Me.customerPhone = value
                End If
            End If
        End Set
    End Property

    ''' <summary>
    ''' The customer zip code or postal code
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property ZipCode() As String
        Get
            Return Me.customerZipCode
        End Get
        Set(ByVal value As String)
            ' Null value?
            If value Is Nothing Then
                Me.customerZipCode = ""
            Else
                ' Only get the first 10 characters
                If (value.Length > 10) Then
                    Me.customerZipCode = value.Substring(0, 10)
                Else
                    Me.customerZipCode = value
                End If
            End If
        End Set
    End Property

    ''' <summary>
    ''' The name of the city in which the customer lives
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property City() As String
        Get
            Return Me.customerCity
        End Get
        Set(ByVal value As String)
            ' Null value?
            If value Is Nothing Then
                Me.customerCity = ""
            Else
                ' Only get the first 30 characters
                If (value.Length > 30) Then
                    Me.customerCity = value.Substring(0, 30)
                Else
                    Me.customerCity = value
                End If
            End If
        End Set
    End Property

    ''' <summary>
    ''' The name of the state or region in which the customer lives
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property State() As String
        Get
            Return Me.customerState
        End Get
        Set(ByVal value As String)
            ' Null value?
            If value Is Nothing Then
                Me.customerState = ""
            Else
                ' Only get the first 30 characters
                If (value.Length > 30) Then
                    Me.customerState = value.Substring(0, 30)
                Else
                    Me.customerState = value
                End If
            End If
        End Set
    End Property

    ''' <summary>
    ''' The ID of the country in which the customer lives
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property CountryID() As Guid?
        Get
            Return Me.customerCountryID
        End Get
        Set(ByVal value As Guid?)
            Me.customerCountryID = value
        End Set
    End Property

    ''' <summary>
    ''' The customer e-mail address
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property EmailAddress() As String
        Get
            Return Me.customerEmailAddress
        End Get
        Set(ByVal value As String)
            If (value Is Nothing) Then
                Me.customerEmailAddress = ""
            Else
                ' Only get the first 40 characters
                If (value.Length > 40) Then
                    Me.customerEmailAddress = value.Substring(0, 40)
                Else
                    Me.customerEmailAddress = value
                End If
            End If
        End Set
    End Property

    ''' <summary>
    ''' The current credit limit of the customer
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property CreditLimit() As Integer
        Get
            Return Me.customerCreditLimit
        End Get
        Set(ByVal value As Integer)
            ' Negative value?
            If value < 0 Then
                Me.customerCreditLimit = 0
            Else
                Me.customerCreditLimit = value
            End If
        End Set
    End Property

    ''' <summary>
    ''' The date the customer was created in the system
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property CreatedDate() As DateTime?
        Get
            Return Me.customerCreatedDate
        End Get
        Private Set(ByVal value As DateTime?)
            ' Date in the past?
            If (value < DateTime.Now) Then
                Me.customerCreatedDate = DateTime.Now
            Else
                Me.customerCreatedDate = value
            End If
        End Set
    End Property

    ''' <summary>
    ''' The name of the user creating the customer
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property CreatedBy() As String
        Get
            Return Me.customerCreatedBy
        End Get
        Set(ByVal value As String)
            ' Null value?
            If value Is Nothing Then
                Me.customerCreatedBy = ""
            Else
                ' Only get the first 15 characters
                If (value.Length > 15) Then
                    Me.customerCreatedBy = value.Substring(0, 15)
                Else
                    Me.customerCreatedBy = value
                End If
            End If
        End Set
    End Property

    ''' <summary>
    ''' The date the customer was last modified in the system
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property ModifiedDate() As DateTime?
        Get
            Return Me.customerModifiedDate
        End Get
        Set(ByVal value As DateTime?)
            ' Date in the past?
            If value < DateTime.Now Then
                Me.customerModifiedDate = DateTime.Now
            Else
                Me.customerModifiedDate = value
            End If
        End Set
    End Property

    ''' <summary>
    ''' The name of the user who last modified the customer
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property ModifiedBy() As String
        Get
            Return Me.customerModifiedBy
        End Get
        Set(ByVal value As String)
            ' Null value?
            If value Is Nothing Then
                Me.customerModifiedBy = ""
            Else
                ' Only get the first 15 characters
                If (value.Length > 15) Then
                    Me.customerModifiedBy = value.Substring(0, 15)
                Else
                    Me.customerModifiedBy = value
                End If
            End If
        End Set
    End Property

    ''' <summary>
    ''' Default parameterless constructor
    ''' </summary>
    ''' <remarks></remarks>
    Public Sub New()
        ' Initialize member backing fields with default values
        Me.ID = Guid.NewGuid()
        Me.CreatedDate = DateTime.Now
    End Sub

    Public Sub New(ByVal id As Guid?)
        ' Initialize member backing fields with 
        ' passed and default values
        Me.ID = id
        Me.CreatedDate = DateTime.Now
    End Sub

    ''' <summary>
    ''' Initializes with a value for all backing fields 
    ''' </summary>
    ''' <param name="id"></param>
    ''' <param name="name"></param>
    ''' <param name="address"></param>
    ''' <param name="phone"></param>
    ''' <param name="zipCode"></param>
    ''' <param name="city"></param>
    ''' <param name="state"></param>
    ''' <param name="countryID"></param>
    ''' <param name="emailAddress"></param>
    ''' <param name="creditLimit"></param>
    ''' <param name="createdDate"></param>
    ''' <param name="createdBy"></param>
    ''' <param name="modifiedDate"></param>
    ''' <param name="modifiedBy"></param>
    ''' <remarks></remarks>
    Public Sub New(ByVal id As Guid?, ByVal name As String, ByVal address As String, _
        ByVal phone As String, ByVal zipCode As String, ByVal city As String, _
        ByVal state As String, ByVal countryID As Guid?, ByVal emailAddress As String, _
        ByVal creditLimit As Integer, ByVal createdDate As DateTime?, _
        ByVal createdBy As String, ByVal modifiedDate As DateTime?, ByVal modifiedBy As String)

        ' Initialize member backing fields with passed values
        Me.ID = id
        Me.CreatedDate = createdDate
        Me.Name = name
        Me.Address = address
        Me.Phone = phone
        Me.ZipCode = zipCode
        Me.City = city
        Me.State = state
        Me.CountryID = countryID
        Me.EmailAddress = emailAddress
        Me.CreditLimit = creditLimit
        Me.CreatedDate = createdDate
        Me.CreatedBy = createdBy
        Me.ModifiedDate = modifiedDate
        Me.ModifiedBy = modifiedBy
    End Sub
End Class